tooling/nargo_fmt/src/formatter/statement.rs

```diff
@@ -800,6 +833,29 @@ mod tests {
         2
     }
 }
+";
+        assert_format(src, expected);
+    }
+
+    #[test]
+    fn format_empty_while() {
+        let src = " fn foo() {  while  condition  {   }  } ";
+        let expected = "fn foo() {
+    while condition {}
+}
+";
+        assert_format(src, expected);
+    }
+
+    #[test]
+    fn format_non_empty_while() {
+        let src = " fn foo() {  while  condition  {  1 ; 2  }  } ";
+        let expected = "fn foo() {
+    while condition {
+        1;
+        2
+    }
```

Comment:
Please add a test for the `If there's a trailing semicolon, remove it` case, e.g.

```suggestion
    }
    
    #[test]
    fn format_while_with_semicolon() {
        let src = " fn foo() {  while  condition  {  1 ; 2  };  } ";
        let expected = "fn foo() {
    while condition {
        1;
        2
    }
```